torch_geometric/transforms/add_metapaths.py

```diff
@@ -100,7 +100,7 @@ def __init__(self, metapaths: List[List[EdgeType]],
         self.keep_same_node_type = keep_same_node_type
         self.drop_unconnected_nodes = drop_unconnected_nodes
 
-    def __call__(self, data: HeteroData) -> HeteroData:
+    def __call__(self, data: HeteroData, max_sample: int = None) -> HeteroData:
```

Comment:
Move to an `__init__` argument.

Reply:
do you think there might be use cases where i use the same AddMetaPath, but want different edge_index with different samples? It might not be a frequent use case, but I just thought it might be more flexible than specifying in __init__.